mcs/class/System.Private.CoreLib/System.Threading/EventWaitHandle.cs

```diff
@@ -1,15 +1,78 @@
+using Microsoft.Win32.SafeHandles;
+using System.Runtime.CompilerServices;
+
 namespace System.Threading
 {
 	partial class EventWaitHandle
 	{
-		public bool Set () => throw new NotImplementedException ();
+		public bool Set ()
+		{
+			SafeWaitHandle handle = ValidateHandle (out bool release);
+
+			try {
+				return SetEventInternal (handle.DangerousGetHandle ());
+			} finally {
+				if (release)
+					handle.DangerousRelease ();
+			}
+		}
+
+		public bool Reset ()
+		{
+			SafeWaitHandle handle = ValidateHandle (out bool release);
+
+			try {
+				return ResetEventInternal (handle.DangerousGetHandle ());
+			} finally {
+				if (release)
+					handle.DangerousRelease ();
+			}
+		}
+
+		unsafe void CreateEventCore (bool initialState, EventResetMode mode, string name, out bool createdNew)
+		{
+			if (name != null)
+				throw new PlatformNotSupportedException (SR.PlatformNotSupported_NamedSynchronizationPrimitives);
+
+			SafeWaitHandle handle = new SafeWaitHandle (CreateEventInternal (mode == EventResetMode.ManualReset, initialState, null, 0, out int errorCode), ownsHandle: true);
+
+			if (errorCode != 0)
+				throw new NotImplementedException ("errorCode");
+
+			createdNew = true;
+		}
+
+		static OpenExistingResult OpenExistingWorker (string name, out EventWaitHandle result)
+		{
+			throw new PlatformNotSupportedException (SR.PlatformNotSupported_NamedSynchronizationPrimitives);
+		}
+
+		internal static bool Set (SafeWaitHandle waitHandle) => throw new NotImplementedException ();
```

Comment:
Would be nice to have this implemented, it is practically identical to the `Set` above.